### ISSUE TITLE
Likehood metric in causalnex.evaluation

### ISSUE THREAD

**Issue by pabloazurduy** [quantumblacklabs/causalnex#85]
Hi, I'm struggling a little bit when evaluating some network alternatives. I know that is possible to get the final node [metrics:](https://causalnex.readthedocs.io/en/latest/source/api_docs/causalnex.evaluation.classification_report.html#causalnex.evaluation.classification_report)

- precision
- recall
- f1-score
- support
- AUC/ROC

the problem is that for an overall score of the network, from the likelihood of a certain sample (or `log-likelihood`) there is no report, I know that probably the solver actually maximizes that number, but I didn't find an easy way to get that number, It will be cool to have a `causalnex.evaluation.likelihood` where we could get the `likelihood(bn, test_data)` so we could compare two or more alternative graphs. (maybe there is an alternative and I'm not aware of..). maybe AUC is a good proxy for the likelihood, (or an overall score) but as far as I know, [they should be different](https://stats.stackexchange.com/questions/422784/does-a-logistic-regression-maximizing-likelihood-necessarily-also-maximize-auc-o)
Many thanks in advance !

**Comment by qbphilip** [comment]
Hello,

I agree, I do not think we report the (log) likelihood. 
quick look:

Adding this method to `structure.pytorch.NotearsMLP` should work
```
    def likelihood(self, x: np.ndarray):
        """
        Args:
            x: 2d numpy array input data, axis=0 is data rows, axis=1 is data columns. 
                Data must be row oriented.
        """
        X_torch = torch.from_numpy(x).float().to(self.device)
        X_hat = self(X_torch)
        for dist_type in self.dist_types:
            loss = loss + dist_type.loss(X_torch, X_hat)
        return loss
```

would you want to evaluate the likelihood for all variables, the target or a subset?

**Comment by qbphilip** [comment]
See #87

**Comment by oentaryorj** [comment]
Closing as per #87
